packages/elements/src/tab-bar/index.ts

```diff
@@ -134,61 +154,147 @@ export class TabBar extends ResponsiveElement {
   }
 
   /**
-   * Hide all scroll buttons
-   * @returns {void}
+   * Return true if tab list have matched value
+   * @param value Value to check
+   * @returns true if tab list have matched value
    */
-  private hideScrollButtons (): void {
-    this.leftBtn.style.setProperty('display', 'none');
-    this.rightBtn.style.setProperty('display', 'none');
+  private isValidValue (value: string) {
+    const tabList = this.getFocusableTabs();
+    return tabList.some(tab => this.getTabValue(tab) === value);
   }
 
   /**
-   * Hide/Show scroll button when element is overflow.
-   * @param elementWidth width of element
+   * When the slot changes, set the level, toggle the scroll button, and set the value
    * @returns {void}
    */
-  private toggleScrollButton (elementWidth: number): void {
-    const { scrollLeft, scrollWidth } = this.content;
-
-    if (this.vertical) {
+  private onSlotChange (): void {
+    this.setLevel();
+    const tabList = this.getFocusableTabs();
+    if (tabList.length < 1) {
       return;
     }
+    const activeTab = tabList.find(tab => tab.active) || tabList[0];
 
-    // handle left button
-    if (scrollLeft > 0) {
-      this.leftBtn.style.setProperty('display', 'flex');
+    if (activeTab) {
+      this.value = this.getTabValue(activeTab);
     }
-    else {
-      this.leftBtn.style.setProperty('display', 'none');
+    // wait element size updated
+    setTimeout(() => {
+      this.toggleScrollButton(this.content.clientWidth);
+    });
+  }
+
+  /**
+   * Mark tab as active
+   * @param value value of tab to select
+   * @returns {void}
+   */
+  private selectValue (value: string): void {
```

Comment:
I think using `active` across the code would be easier to navigate, using `select` may cause confusion with the element state. Something like `activateTab`, `setActiveTab` etc. would sound better.